src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,432 @@
+/* Copyright 2013-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace MongoDB.Driver.Core.Misc
+{
+    /// <summary>
+    /// Represents a server version number.
+    /// </summary>
+    public class ServerVersion : IEquatable<ServerVersion>, IComparable<ServerVersion>
+    {
+        // fields
+        private readonly string _commitHash;
```

Comment:
Commit hash (`b8f58d7` part of `4.5.0-489-gb8f58d7`, `g` is from git describe) is present for round trip purpose only. I mean if we parse a `4.5.0-489-gb8f58d7` and then execute `.ToString()`, we will get the same string. Commit hash is not strictly essential for version comparison.